src/Proto3/Suite/DotProto/Internal.hs

```diff
@@ -391,17 +392,35 @@ toPascalCase xs = foldMap go (segmentBy (== '_') xs)
 
 -- | @'toCamelCase' xs@ sends a snake-case string @xs@ to a camel-cased string.
 toCamelCase :: String -> String
-toCamelCase xs = case toPascalCase xs of
-  "" -> ""
-  x : xs' -> toLower x : xs'
+toCamelCase xs =
+  case toPascalCase xs of
+    "" -> ""
+    x : xs' -> toLower x : xs'
 
--- | @'toUpperFirst' xs@ sends the first character @x@ in @xs@ to be upper case, @'toUpperFirst' "" = ""@.
+-- | Uppercases the first character of a string.
+--
+-- ==== __Examples__
+--
+-- >>> toUpperFirst "abc"
+-- "Abc"
+--
+-- >>> toUpperFirst ""
+-- ""
 toUpperFirst :: String -> String
-toUpperFirst [] = []
+toUpperFirst "" = ""
 toUpperFirst (x : xs) = toUpper x : xs
```

Comment:
`typeLikeName`, `fieldLikeName`, `prefixedMethodName` and `toCamelCase` all use the quote pattern for matching on the empty string so I was trying to make `toUpperFirst` consistent with those functions. I prefer the quote pattern because it stands out that we're matching on a string rather than just a list. 

I guess I could see an argument against it with `-XOverloadedString`, but `[]` isn't monomorphic either. My preference here isn't super strong. I can replace usages of the pattern with empty list pattern if needed.

Reply:
My point about polymorphism is just that `""` by itself does not tell you that the type is `String`.  But you make a good point about other functions using `""` in the same way, so this is then a good change because it brings uniformity.